src/main/resources/controllers-context.xml

```diff
@@ -65,4 +60,18 @@
         </mvc:message-converters>
     </mvc:annotation-driven>
 
+    <beans profile="transactionalDogService">
+        <bean id="transactionalDogService" class="com.example.dog.service.TransactionalDogService">
```

Comment:
The purpose of these profiles is to replace jdbcDynamicProxy with the manual proxy or vice versa by switching the profiles. If you have different IDs of beans, you won't be able to accomplish that.